### PR TITLE
Added APIs for Async Submission CSV Export

### PR DESCRIPTION
**Pull request by nialldonnellyfh**
The user has the option to add the `--async` parameter to start an asynchronous CSV Export process.

This is useful for long-running CSV export operations.

The existing synchronous API still works as before.

Published to skunkhenry as `fh-fhc-2.11.0-370`

### CODE REVIEW

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* The code changes conform to the contributing & code style guides for this component (if they exist)
* A component owner should review your change (we've tagged these above)
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* If updating server component this PR should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Ensure JIRA is up to date before & after merge (e.g. deployment notes, tester notes, release notes, fix version)
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.
Fhcap PR will be generated automatically.
To deploy your changes to a cluster place comment in this PR with pattern [x][y] where x=deploy and y=clustername you wish to deploy. 
You can also deploy your change manually using fhcap-cli

**Comment by vchepeli** [comment]
LGTM

File: lib/cmd/fh3/appforms/environments/submissions/exportcsv.js [hunk 1]
```diff
@@ -2,11 +2,85 @@
...
+    fhreq.GET(undefined, url, function(err, asyncStatus) {
+      asyncStatus = asyncStatus || {};
+      if(err) {
+        clearInterval(csvExportPollInterval);
+        return cb(err);
+      }
+
+      if(asyncStatus.status === 'inprogress') {
+        //Just updating the user on the status of the export
+        log.info(asyncStatus.message);
```
Comment (paolobueno) [inline lib/cmd/fh3/appforms/environments/submissions/exportcsv.js]: This `message` is something like "exporting submission #x of y", right? Is there a way to include more fields on the status or parsing the message to show a progress bar? We've added one for the polling on the appdata export here: https://github.com/feedhenry/fh-fhc/blob/master/lib/cmd/fh3/appdata/export/status.js#L32

**Comment by paolobueno** [comment]
LGTM, one suggestion to improve UX, but not critical.

File: lib/cmd/fh3/appforms/environments/submissions/exportcsv.js (hunk 1 above)
Comment (nialldonnellyfh) [inline lib/cmd/fh3/appforms/environments/submissions/exportcsv.js]: Not all of them. There are messages around building metadata etc in between

Comment (paolobueno) [inline lib/cmd/fh3/appforms/environments/submissions/exportcsv.js]: Looks like you can display them through [custom tokens](https://github.com/visionmedia/node-progress#custom-tokens)

Comment (nialldonnellyfh) [inline lib/cmd/fh3/appforms/environments/submissions/exportcsv.js]: For the moment, the messages are fine I think, but will add a progress bar later when I can change the progress update data in fh-forms.

File: lib/cmd/fh3/appforms/environments/submissions/exportcsv.js [hunk 2]
```diff
@@ -2,11 +2,85 @@
...
+        return cb(err);
+      }
+
+      if(asyncStatus.status === 'inprogress') {
+        //Just updating the user on the status of the export
+        console.log(asyncStatus.message);
+      } else if(asyncStatus.status === 'complete') {
+        //Export is complete, can now download the file.
+        clearInterval(csvExportPollInterval);
+        console.log("Submission CSV export complete. Downloading the exported file.");
```
Comment (fh-qube) [inline lib/cmd/fh3/appforms/environments/submissions/exportcsv.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

File: lib/cmd/fh3/appforms/environments/submissions/exportcsv.js [hunk 3]
```diff
@@ -2,11 +2,85 @@
...
+    fhreq.GET(undefined, url, function(err, asyncStatus) {
+      asyncStatus = asyncStatus || {};
+      if(err) {
+        clearInterval(csvExportPollInterval);
+        return cb(err);
+      }
+
+      if(asyncStatus.status === 'inprogress') {
+        //Just updating the user on the status of the export
+        console.log(asyncStatus.message);
```
Comment (fh-qube) [inline lib/cmd/fh3/appforms/environments/submissions/exportcsv.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: lib/cmd/fh3/appforms/environments/submissions/exportcsv.js [hunk 4]
```diff
@@ -2,11 +2,92 @@
...
+        return cb(err);
+      }
+
+      if(asyncStatus.status === 'inprogress') {
+        //Just updating the user on the status of the export
+
+        //Only log out a progress message if it has changed. Otherwise it is too verbose.
+        if(asyncStatus.message !== prevMessage) {
+          prevMessage = asyncStatus.message;
+          console.log(asyncStatus.message);
```
Comment (fh-qube) [inline lib/cmd/fh3/appforms/environments/submissions/exportcsv.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

File: lib/cmd/fh3/appforms/environments/submissions/exportcsv.js [hunk 5]
```diff
@@ -2,11 +2,92 @@
...
+        }
+      } else if(asyncStatus.status === 'complete') {
+        //Export is complete, can now download the file.
+        clearInterval(csvExportPollInterval);
+        console.log("Submission CSV export complete. Downloading the exported file.");
```
Comment (fh-qube) [inline lib/cmd/fh3/appforms/environments/submissions/exportcsv.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Pull request merged**